public/app/core/components/MegaMenu/NavBarMenu.tsx

```diff
@@ -27,56 +28,74 @@ export function NavBarMenu({ activeItem, navItems, searchBarHidden, onClose }: P
   const styles = getStyles(theme, searchBarHidden);
   const animationSpeed = theme.transitions.duration.shortest;
   const animStyles = getAnimStyles(theme, animationSpeed);
+  const { chrome } = useGrafana();
+  const state = chrome.useState();
   const ref = useRef(null);
+  const backdropRef = useRef(null);
   const { dialogProps } = useDialog({}, ref);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const onMenuClose = () => setIsOpen(false);
 
   const { overlayProps, underlayProps } = useOverlay(
     {
       isDismissable: true,
       isOpen: true,
-      onClose,
+      onClose: onMenuClose,
     },
     ref
   );
 
+  useEffect(() => {
+    if (state.megaMenuOpen) {
+      setIsOpen(true);
+    }
+  }, [state.megaMenuOpen]);
+
   return (
     <OverlayContainer>
-      <FocusScope contain autoFocus>
-        <CSSTransition appear={true} in={true} classNames={animStyles.overlay} timeout={animationSpeed}>
-          <div data-testid="navbarmenu" ref={ref} {...overlayProps} {...dialogProps} className={styles.container}>
-            <div className={styles.mobileHeader}>
-              <Icon name="bars" size="xl" />
-              <IconButton
-                aria-label="Close navigation menu"
-                name="times"
-                onClick={onClose}
-                size="xl"
-                variant="secondary"
-              />
-            </div>
-            <NavBarToggle
-              className={styles.menuCollapseIcon}
-              isExpanded={true}
-              onClick={() => {
-                reportInteraction('grafana_navigation_collapsed');
-                onClose();
-              }}
+      <CSSTransition
+        in={isOpen}
+        unmountOnExit={true}
+        classNames={animStyles.overlay}
+        timeout={animationSpeed}
+        onExited={onClose}
+      >
+        <div data-testid="navbarmenu" ref={ref} {...overlayProps} {...dialogProps} className={styles.container}>
+          <div className={styles.mobileHeader}>
+            <Icon name="bars" size="xl" />
+            <IconButton
+              aria-label="Close navigation menu"
+              name="times"
+              onClick={onMenuClose}
+              size="xl"
+              variant="secondary"
             />
+          </div>
+          <NavBarToggle
+            className={styles.menuCollapseIcon}
+            isExpanded={true}
+            onClick={() => {
+              reportInteraction('grafana_navigation_collapsed');
+              onMenuClose();
+            }}
+          />
```

Comment:
don't think we can move this outside of the `FocusScope` as there's then no tabbable element to focus which will close the menu 🤔